example/__tests__/videoroomContext.test.ts

```diff
@@ -108,4 +129,68 @@ describe('Videoroom context', () => {
     expect(useAudioTrackMetadataMock).toBeCalledTimes(1);
     expect(result.current.isMicrophoneOn).toBe(false);
   });
+
+  test('Connecting to room', async () => {
+    const { result } = renderHook(() => useVideoroomState(), {
+      wrapper: VideoroomContextProvider,
+    });
+
+    act(() => {
+      result.current.setRoomName('test room     ');
+      result.current.setUsername('test user   ');
+    });
+    await act(async () => {
+      await result.current.connectAndJoinRoom();
+    });
+
+    expect(result.current.roomName).toBe('test room');
+    expect(result.current.username).toBe('test user');
+    expect(setExtra).toBeCalledTimes(2);
+    expect(connectCallback).toBeCalledTimes(1);
+    expect(joinCallback).toBeCalledTimes(1);
+    expect(result.current.videoroomState).toBe('InMeeting');
+  });
+
+  test('Disconnect from room', async () => {
+    const { result } = renderHook(() => useVideoroomState(), {
+      wrapper: VideoroomContextProvider,
+    });
+
+    await act(async () => {
+      await result.current.connectAndJoinRoom();
+      await result.current.disconnect();
+    });
+
+    expect(disconnectCallback).toBeCalledTimes(1);
+    expect(result.current.videoroomState).toBe('AfterMeeting');
```

Comment:
I'd add a test for rejoining, it's a bit tricky. Also a test for going back to the main screen (the name and room name are reset).

Reply:
Added rejoining scenario. 
There was already a test checking for going back to the main screen behaviour. I updated it to call `disconnect` before leaving. I think that's enough given that those are supposed to be unit tests only.